lib/src/ansi_code.dart

```diff
@@ -0,0 +1,286 @@
+// Copyright 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io' as io;
+
+/// The type of code represented by [AnsiCode].
+class AnsiCodeType {
+  final String _name;
+
+  /// A foreground color.
+  static const AnsiCodeType foreground = const AnsiCodeType._('foreground');
+
+  /// A style.
+  static const AnsiCodeType style = const AnsiCodeType._('style');
+
+  /// A background color.
+  static const AnsiCodeType background = const AnsiCodeType._('background');
+
+  /// A reset value.
+  static const AnsiCodeType reset = const AnsiCodeType._('reset');
+
+  const AnsiCodeType._(this._name);
+
+  @override
+  String toString() => 'AnsiType.$_name';
+}
+
+/// Standard ANSI escape code for customizing terminal text output.
+///
+/// [Source](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors)
+class AnsiCode {
+  /// The numeric value associated with this code.
+  final int code;
+
+  /// The [AnsiCode] that resets this value, if one exists.
+  ///
+  /// Otherwise, `null`.
+  final AnsiCode reset;
+
+  /// A description of this code.
+  final String name;
+
+  /// The type of code that is represented.
+  final AnsiCodeType type;
+
+  const AnsiCode._(this.name, this.type, this.code, this.reset);
+
+  /// Represents the value escaped for use in terminal output.
+  String get escape => "\x1B[${code}m";
+
+  /// Wraps [value] with the [escape] value for this code, followed by
+  /// [resetAll].
+  ///
+  /// Returns `value` unchanged if
+  ///   * [value] is `null` or empty
+  ///   * `dart:io` `stdout.supportsAnsiEscapes` is `false`
+  ///   * [type] is [AnsiCodeType.reset]
+  String wrap(String value) => (io.stdout.supportsAnsiEscapes &&
```

Comment:
Probably not a big deal, but it would be nice if there were a way to tell it which `StdOut` stream to check for compatibility. For instance maybe you are piping stdout to a file so this would be false, but stderr is still going to the console and you would like nice looking errors there.